app/javascript/mastodon/features/compose/components/emoji_picker_dropdown.js

```diff
@@ -54,6 +61,24 @@ class EmojiPickerDropdown extends React.PureComponent {
     this.props.onPickEmoji(data);
   }
 
+  onShowDropdown () {
+    this.setState({active: true});
+    if (!EmojiPicker) {
+      this.setState({loading: true});
+      import('emojione-picker').then(TheEmojiPicker => {
```

Comment:
The [dynamic import proposal](https://github.com/tc39/proposal-dynamic-import) is at stage-3 in TC39 so I feel pretty confident betting on this. [The Webpack docs also recommend it](https://webpack.js.org/guides/code-splitting-async/) over the legacy `require.ensure()`.